core/isolate.rs

```diff
@@ -55,6 +58,96 @@ pub type OpResult<E> = Result<Op<E>, E>;
 /// Args: op_id, control_buf, zero_copy_buf
 type CoreDispatchFn = dyn Fn(OpId, &[u8], Option<PinnedBuf>) -> CoreOp;
 
+pub trait OpDispatcher: Send + Sync {
+    fn dispatch(&self, args: &[u8], buf: Option<PinnedBuf>) -> CoreOp;
+}
+
+// Use a separate trait to give the op a name, because a trait object
+// object (dyn Opdispatcher) can't have associated constants.
+pub trait NamedOpDispatcher: OpDispatcher {
+    const NAME: &'static str;
+}
+
+pub struct DispatcherRegistry {
```

Comment:
Maybe move this to a separate file, core/op_dispatchers.rs  (or however you want to name it)

Reply:
👍 Also that name really bothers me. Maybe `OpReg`?

Reply:
Sure